app/src/main/java/org/dharmaseed/android/PlayTalkActivity.java

```diff
@@ -139,58 +162,149 @@ protected void onCreate(Bundle savedInstanceState) {
         // Set date
         TextView dateView = (TextView) findViewById(R.id.play_talk_date);
         String recDate = talk.getDate();
-        SimpleDateFormat parser = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        SimpleDateFormat parser = new SimpleDateFormat(DATE_FORMAT);
         try {
             dateView.setText(DateFormat.getDateInstance().format(parser.parse(recDate)));
-        } catch(ParseException e) {
+        } catch (ParseException e) {
             dateView.setText("");
             Log.w(LOG_TAG, "Could not parse talk date for talk ID " + talkID);
         }
 
+        // Initialise seek bar
+        final SeekBar seekBar = (SeekBar) findViewById(R.id.play_talk_seek_bar);
+        seekBar.setMax((int)(60*1000*talk.getDurationInMinutes()));
+        userDraggingSeekBar = false;
+        seekBar.setOnSeekBarChangeListener(this);
+
         // Get/create a persistent fragment to manage the MediaPlayer instance
         FragmentManager fm = getSupportFragmentManager();
         talkPlayerFragment = (TalkPlayerFragment) fm.findFragmentByTag("talkPlayerFragment");
         if (talkPlayerFragment == null) {
             // add the fragment
             talkPlayerFragment = new TalkPlayerFragment();
             fm.beginTransaction().add(talkPlayerFragment, "talkPlayerFragment").commit();
+            prepareTalkPlayerFragment(talkPlayerFragment);
+
+            // retrieve progress from TalkHistory DB table
+            final int pos = (int) (dbManager.getTalkProgress(talkID)*60*1000);
+            setTalkProgress(pos, false);
         } else if(talkPlayerFragment.getMediaPlayer().isPlaying()) {
             setPPButton("ic_media_pause");
+            updateSeekBar();
+            Log.i(LOG_TAG,"talk "+talkID+" already playing!");
         }
 
-        // Set the talk duration
-        final TextView durationView = (TextView) findViewById(R.id.play_talk_talk_duration);
-        double duration = talk.getDurationInMinutes();
-        String durationStr = DateUtils.formatElapsedTime((long)(duration*60));
-        durationView.setText(durationStr);
+        // initialise timers
+        timer = new Timer();
```

Comment:
Thanks for noticing the problem with using `Handler` and updating it to use `Timer` - I just learned that exists thanks to your PR. :wink: